### ISSUE TITLE
Missing documentation on cache invalidation

### ISSUE THREAD

**Issue by RudeySH** [moozzyk/EFCache#30]
It is unclear how this library handles cache invalidation. Can we set a timeout? Can we invalidate cache based on keys? Does it automatically invalidate cache when updating entities using EF?

**Comment by RudeySH** [comment]
I see, thanks! If I understood correctly, are you saying cache is invalidated regardless of whether or not a write actually impacted the cached results, as long as the write affected table(s) that a cached result was retrieved from? For example, if I filter records with `WHERE Prop = 'A'` and cache that, will my cache be invalidated when a record with `Prop = 'B'` is inserted, updated or deleted?

**Comment by moozzyk** [comment]
This is correct. I don't think there is an easy way with EF to tell which rows are cached vs. which rows will be affected by a CUD command (and also queries are sometimes super complicated - especially where hierarchies are involved) so EFCache operates on table granularity. This is one of the reasons I always recommend to look at what is actually worth caching and create a specialized caching policy to cache results for queries that don't involve tables that are constantly modified. If you don't care about expiration there is a way of achieving the same result without specialized caching policy - `CachingPolicy` has a constructor that allows you to pass a list of tables whose results are subject to caching. This is the easiest way to limit what you want to cache.
